pkg/util/timerecord/group_checker.go

```diff
@@ -75,18 +99,17 @@ func (gc *GroupChecker) work() {
 	}
 }
 
-// Check updates the latest timestamp for provided name
-func (gc *GroupChecker) Check(name string) {
-	gc.lastest.Insert(name, time.Now())
+func (gc *CheckerManager) Register(name string, checker *Checker) {
+	gc.checkers.Insert(name, checker)
```

Comment:
It looks like we always create a new one. Should we check if name already exists?